### DISCUSSION TITLE
How to make "Show chatgpt-output" wait until finished?

### DISCUSSION THREAD

**Discussion by vinta** [Yue-Yang/ChatGPT-Siri#60] [Ideas]
It's often that Siri is still speaking the content of `chatgpt-output`, but the shortcut is showing the "continue-text".

**Comment by Yue-Yang** [comment]
There is no other way. The operations of shortcuts are limited. Another possible approach is to replace "show" with Siri reading the text aloud. However, this would require complex settings and could bring other problems, such as Siri reading out the text when users click on it. Currently, there is little room for improvement in my version, as I still need to consider "click to use" and cross-platform compatibility.

**Reply by vinta** [reply]
I see. Then just some random idea: maybe I can get the length of `chatgpt-output`, and roughly calculate how many seconds Siri needs to speak the whole content, and wait for xxx seconds after the "Show chatgpt-output" action?

Thank you for your reply! This shortcut is fascinating!
